File: sample/src/main/java/com/example/kyrie/SampleListenerAdapter.java
package com.example.kyrie;

import android.widget.SeekBar;

import androidx.annotation.NonNull;

import com.github.alexjlockwood.kyrie.KyrieDrawable;

final class SampleListenerAdapter extends KyrieDrawable.ListenerAdapter {
  @NonNull private final SeekBar seekBar;

  public SampleListenerAdapter(@NonNull SeekBar seekBar) {
    this.seekBar = seekBar;
  }

  @Override
  public void onAnimationUpdate(@NonNull KyrieDrawable drawable) {
    final float playTime = drawable.getCurrentPlayTime();
    final float totalDuration = drawable.getTotalDuration();
    final float fraction = playTime / totalDuration;
    seekBar.setProgress(Math.round(fraction * seekBar.getMax()));
  }
}

File: sample/src/main/java/com/example/kyrie/SampleListenerAdapter.kt
package com.example.kyrie

import android.widget.SeekBar

import com.github.alexjlockwood.kyrie.KyrieDrawable

internal class SampleListenerAdapter(private val seekBar: SeekBar) : KyrieDrawable.ListenerAdapter() {

    override fun onAnimationUpdate(drawable: KyrieDrawable) {
        val playTime = drawable.currentPlayTime.toFloat()
        val totalDuration = drawable.totalDuration.toFloat()
        val fraction = playTime / totalDuration
        seekBar.progress = Math.round(fraction * seekBar.max)
    }
}
